hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMNodeDetails.java

```diff
@@ -17,59 +17,61 @@
 
 package org.apache.hadoop.hdds.scm.ha;
 
-import org.apache.hadoop.hdds.conf.OzoneConfiguration;
-import org.apache.hadoop.hdds.scm.ScmConfigKeys;
 import org.apache.hadoop.net.NetUtils;
+import org.apache.hadoop.ozone.ha.NodeDetails;
+import org.apache.ratis.protocol.RaftGroup;
+import org.apache.ratis.protocol.RaftPeerId;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.io.IOException;
-import java.net.InetAddress;
 import java.net.InetSocketAddress;
 
-import static org.apache.hadoop.hdds.scm.ScmConfigKeys.OZONE_SCM_INTERNAL_SERVICE_ID;
-import static org.apache.hadoop.hdds.scm.ScmConfigKeys.OZONE_SCM_SERVICE_IDS_KEY;
-
 /**
  * Construct SCM node details.
  */
-public final class SCMNodeDetails {
-  private String scmServiceId;
-  private String scmNodeId;
-  private InetSocketAddress rpcAddress;
-  private int rpcPort;
-  private int ratisPort;
-  private String httpAddress;
-  private String httpsAddress;
+public final class SCMNodeDetails extends NodeDetails {
+  private InetSocketAddress blockProtocolServerAddress;
+  private String blockProtocolServerAddressKey;
+  private InetSocketAddress clientProtocolServerAddress;
+  private String clientProtocolServerAddressKey;
+  private InetSocketAddress datanodeProtocolServerAddress;
+  private String datanodeAddressKey;
 
   public static final Logger LOG =
       LoggerFactory.getLogger(SCMNodeDetails.class);
 
   /**
    * Constructs SCMNodeDetails object.
    */
+  @SuppressWarnings("checkstyle:ParameterNumber")
   private SCMNodeDetails(String serviceId, String nodeId,
-                        InetSocketAddress rpcAddr, int rpcPort, int ratisPort,
-                        String httpAddress, String httpsAddress) {
-    this.scmServiceId = serviceId;
-    this.scmNodeId = nodeId;
-    this.rpcAddress = rpcAddr;
-    this.rpcPort = rpcPort;
-    this.ratisPort = ratisPort;
-    this.httpAddress = httpAddress;
-    this.httpsAddress = httpsAddress;
+      InetSocketAddress rpcAddr, int ratisPort, String httpAddress,
+      String httpsAddress, InetSocketAddress blockProtocolServerAddress,
+      InetSocketAddress clientProtocolServerAddress,
+      InetSocketAddress datanodeProtocolServerAddress,
+      RaftGroup group, RaftPeerId selfPeerId, String datanodeAddressKey,
+      String blockProtocolServerAddressKey,
+      String clientProtocolServerAddressAddressKey) {
+    super(serviceId, nodeId, rpcAddr, ratisPort,
+        httpAddress, httpsAddress);
+    this.blockProtocolServerAddress = blockProtocolServerAddress;
+    this.clientProtocolServerAddress = clientProtocolServerAddress;
+    this.datanodeProtocolServerAddress = datanodeProtocolServerAddress;
+    this.datanodeAddressKey = datanodeAddressKey;
+    this.blockProtocolServerAddressKey = blockProtocolServerAddressKey;
+    this.clientProtocolServerAddressKey = clientProtocolServerAddressAddressKey;
   }
 
   @Override
   public String toString() {
+    // TODO: add new fields to toString
```

Comment:
Can we move this to NodeDetails class itself?

Reply:
Probably not. 

SCMNodeDetails and OMNodeDetails do not share the same fields. E.g. OMNodeDetails uses RPC port while SCMNodeDetails has three different server port.